### PR TITLE
Doctests for pronoun frequency analysis

### PR DESCRIPTION
**Pull request by kmerrill18**
I added doctests for the analysis methods in gender pronoun freq analysis with the test_corpus of ten books.

### CODE REVIEW

**Comment by srisi** [comment]
Thanks for the additions.
I'm not quite sure why your pickled files are so large but for right now, we can't merge them. 
There are two options:
1: Track down why the pickled files are so large. I'll also have a quick look. 
2: Add the files to .gitignore so they are only stored on your local machine but not in your git.

File: gender_novels/analysis/gender_pronoun_freq_analysis.py [hunk 1]
```diff
@@ -34,21 +44,33 @@ def books_pronoun_freq(corp):
         relative_freq_male[book] = temp_dict['male']
```
Comment (srisi) [inline gender_novels/analysis/gender_pronoun_freq_analysis.py]: I'm confused by the size of the dict. Maybe using book as the key makes the whole book the key? I.e. including the whole text.
You could try `relative_freq_male[str(book)] = temp_dict['male']` to see if that reduces the filesize.

**Comment by kmerrill18** [comment]
I use the novel object as the key because I want to be able to get attributes about the novels (like author gender, location, etc.) later for analysis. There is a different way to write it that uses subcorpuses with the attribute you want already sorted which would save space, but we couldn't do that for most of the project since there was no subcorpus method until recently. I guess we could rewrite it now but that seems like a lot of work for what it's worth at this point. I'll just store the pickle files on my machine if that's okay.
________________________________

**Comment by srisi** [comment]
> I use the novel object as the key because I want to be able to get attributes about the novels (like author gender, location, etc.) later for analysis.

Ah, I see. 

If you want to keep the files on your disk, make sure to add them to .gitignore. You'll probably also either have to delete them (or move them somewhere else), commit, and then re-run the analysis (or copy the files back). Beware though that the gutenberg files are going to be 40 times larger.
One alternative (assuming you only need the metadata but not the text) would be to delete the text and wordcounter before storing them.
```
        book.text = ''
        book._word_counts_counter = None

        relative_freq_male[book] = temp_dict['male']
        relative_freq_female[book] = temp_dict['female']
```
That way, the size drops almost to zero but you still have access to author, gender, place etc.

**Comment by kmerrill18** [comment]
Thank you! Will do!
________________________________

**Comment by srisi** [comment]
> Thank you! Will do!

Thanks. The updated zip files look good! 
However, your whole repo is a few commits behind the upstream master and right now your pull request is trying to undo some of the work from the last day. Please fetch and merge updates from upstream and then update the pull request. Thanks.

**Comment by kmerrill18** [comment]
Github was doing the thing where it wouldn't let me merge so I force fixed it and now I have to re-write everything. Ahhh! Anyway, it will be a minute, but I'll get it.
________________________________

**Pull request closed without merging**